sql_csv_utils.py

```diff
@@ -16,7 +16,16 @@ def __init__(self, config, logging_level=logging.INFO):
         self.specify_db_connection = SpecifyDb(db_config_class=self.config)
         self.logger = logging.getLogger("SqlCsvTools")
         self.logger.setLevel(logging_level)
-        self.specify_db_connection.connect()
+        self.check_db_connection()
+
+    def check_db_connection(self):
+        """checking whether database connection is functional"""
+        try:
+            self.specify_db_connection.connect()
+            self.logger.info("sql_csv_tools connection established")
+        except Exception as e:
+            raise ValueError(f"Database connection error: {e}")
```

Comment:
You can create a custom exception by inherting from the "Exception" class. "valueException" is a poor choice; I'd say create a custom "DatabaseConnectionError". there are examples of this in the code.

Reply:
ok, will do